File: library/src/main/java/me/proxer/library/api/messenger/ConferenceInfoEndpoint.java
package me.proxer.library.api.messenger;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;
import me.proxer.library.entitiy.messenger.Conference;
import me.proxer.library.entitiy.messenger.ConferenceInfo;

/**
 * Endpoint for retrieving information concerning a {@link Conference}.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class ConferenceInfoEndpoint implements Endpoint<ConferenceInfo> {

    private final InternalApi internalApi;

    private final String id;

    ConferenceInfoEndpoint(final InternalApi internalApi, final String id) {
        this.internalApi = internalApi;
        this.id = id;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<ConferenceInfo> build() {
        return internalApi.conferenceInfo(id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/messenger/ConferenceInfoEndpoint.kt
package me.proxer.library.api.messenger

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint
import me.proxer.library.entity.messenger.Conference
import me.proxer.library.entity.messenger.ConferenceInfo

/**
 * Endpoint for retrieving information concerning a [Conference].
 *
 * @author Ruben Gees
 */
class ConferenceInfoEndpoint internal constructor
    (
    private val internalApi: InternalApi,
    private val id: String
) : Endpoint<ConferenceInfo> {

    override fun build(): ProxerCall<ConferenceInfo> {
        return internalApi.conferenceInfo(id)
    }
}
